cmd/cat/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	var lines int
+	flag.IntVar(&lines, "n", -1, "custom number of lines to print")
+	flag.Parse()
+
+	var fileName string
+	var input io.Reader
+
+	if len(os.Args) == 1 {
+		fmt.Println("No file passed")
+		os.Exit(1)
+	} else if len(os.Args) == 2 {
+		fileName = os.Args[1]
+	} else if len(os.Args) == 4 {
+		fileName = os.Args[3]
+	}
+
+	file, err := os.Open(fileName)
+	if err != nil {
+		fmt.Println("Cannot open file!")
+		os.Exit(1)
+	}
+
+	input = file
+
+	scanner := bufio.NewScanner(input)
+	if lines == -1 {
```

Comment:
I don't think you implemented the right -n flag, please check its behavior and check your implementation

Reply:
> I don't think you implemented the right -n flag, please check its behavior and check your implementation

doesn't cat print the first n lines of a file?